.ddev/config.yaml

```
@@ -0,0 +1,296 @@
+name: matomo
+type: php
+docroot: ""
+php_version: "8.3"
+webserver_type: apache-fpm
+router_http_port: "80"
+router_https_port: "443"
+xdebug_enabled: false
+additional_hostnames: []
+additional_fqdns: []
+database:
+    type: mysql
+    version: "5.7"
+hooks:
+    post-start:
+        - exec: MYSQL_PWD=root mysql -uroot -e "CREATE DATABASE IF NOT EXISTS matomo_tests; GRANT ALL ON matomo_tests.* to 'db'@'%';"
+          service: db
+        - exec: MYSQL_PWD=root mysql -uroot -e "CREATE DATABASE IF NOT EXISTS _UITestFixture; GRANT ALL ON _UITestFixture.* to 'db'@'%';"
+          service: db
```

```suggestion
        - exec: MYSQL_PWD=root mysql -uroot -e "GRANT ALL PRIVILEGES ON *.* to 'db'@'%'"
          service: db
        - exec: MYSQL_PWD=root mysql -uroot -e "GRANT ALL PRIVILEGES ON *.* to 'db'@'localhost'"
          service: db
```

This would allow us to configure `db:db` for the tests instead of `root:root`.

Otherwise there may be databases not visible to the `db` user, for example when running when running the `EmptySite` spec:

```shell
ddev matomo:console tests:run-ui --persist-fixture-data EmptySite
```

Without more privileges the created `_EmptySite` database would not be visible.

_Not 100% sure if we need to run the statement for if for `@localhost`, I think it was required to have both `ddev mysql` and `mysql` to have all the privileges.._

---

I wouldn't expect the `@localhost` to be needed as the db is not running on the same system so PHP will never access it on localhost.

I'll test it out, thanks for pointing this out and identifying a test that can be impacted by this.